pkg/image/pull_policy.go

```diff
@@ -1,41 +1,312 @@
+
+	duration, err := parseDurationString(durationStr)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse duration from JSON")
+	}
+
+	timeThreshold := time.Now().Add(-duration)
+
+	return imageTimestamp.Before(timeThreshold), nil
+}
+
+func parseDurationString(durationStr string) (time.Duration, error) {
+	var totalMinutes int
+	for i := 0; i < len(durationStr); {
+		endIndex := i + 1
+		for endIndex < len(durationStr) && durationStr[endIndex] >= '0' && durationStr[endIndex] <= '9' {
+			endIndex++
+		}
+
+		value, err := strconv.Atoi(durationStr[i:endIndex])
+		if err != nil {
+			return 0, errors.Wrapf(err, "invalid interval format: %s", durationStr)
+		}
+		unit := durationStr[endIndex]
+
+		switch unit {
+		case 'd':
+			totalMinutes += value * 24 * 60
+		case 'h':
+			totalMinutes += value * 60
+		case 'm':
+			totalMinutes += value
+		default:
+			return 0, errors.Errorf("invalid interval uniit: %s", string(unit))
+		}
+
+		i = endIndex + 1
+	}
+
+	return time.Duration(totalMinutes) * time.Minute, nil
+}
+
+func (f *Fetcher) PruneOldImages() error {
+	imageJSON, err := ReadImageJSON(f.logger)
+	if err != nil {
+		return err
+	}
+
+	if imageJSON.Interval.LastPrune != "" {
+		lastPruneTime, err := time.Parse(time.RFC3339, imageJSON.Interval.LastPrune)
+		if err != nil {
+			return errors.Wrap(err, "failed to parse last prune timestamp from JSON")
+		}
+
+		pruningInterval, err := parseDurationString(imageJSON.Interval.PruningIinterval)
+		if err != nil {
+			return errors.Wrap(err, "failed to parse pruning interval from JSON")
+		}
+
+		if time.Since(lastPruneTime) < pruningInterval {
+			// not enough time has passed since the last prune
+			return nil
+		}
+	}
+
+	// prune images older than the pruning interval
+	pruningDuration, err := parseDurationString(imageJSON.Interval.PruningIinterval)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse pruning interval from JSON")
+	}
+
+	pruningThreshold := time.Now().Add(-pruningDuration)
+
+	for imageID, timestamp := range imageJSON.Image.ImageIDtoTIME {
+		imageTimestamp, err := time.Parse(time.RFC3339, timestamp)
+		if err != nil {
+			return errors.Wrap(err, "failed to parse image timestamp fron JSON")
+		}
+
+		_, err = f.fetchDaemonImage(imageID)
+		if !errors.Is(err, ErrNotFound) {
+			delete(imageJSON.Image.ImageIDtoTIME, imageID)
+		}
+
+		if imageTimestamp.Before(pruningThreshold) {
+			delete(imageJSON.Image.ImageIDtoTIME, imageID)
+		}
+	}
+
+	imageJSON.Interval.LastPrune = time.Now().Format(time.RFC3339)
+
+	updatedJSON, err := json.MarshalIndent(imageJSON, "", "    ")
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal updated records")
+	}
+
+	if err := WriteFile(updatedJSON); err != nil {
+		return errors.Wrap(err, "failed to write updated image.json")
+	}
+
+	return nil
+}
+
+func WriteFile(data []byte) error {
+	cmd := exec.Command("sudo", "sh", "-c", "echo '"+string(data)+"' > "+imagePath)
```

Comment:
@jjbustamante I was trying it like this - 
```
func WriteFile(data []byte, path string) error {
	file, err := os.OpenFile(path, os.O_WRONLY|os.O_TRUNC, 0644)
	if err != nil {
		return errors.New("failed to open file: " + err.Error())
	}
	defer file.Close()

	_, err = file.Write(data)
	if err != nil {
		return errors.New("failed to write data to file: " + err.Error())
	}

	return nil
}
```
but there is some kind of bug, it can't detect the actual path (`$HOME/.pack/image.json`) of `image.json` it's trying to write the file under `/root/.pack/ `